### PR TITLE
Debian packaging

### PR DESCRIPTION
**Pull request by luis-lavaire**

### CODE REVIEW

**Comment by azubieta** [comment]
The idea is not to drop the Appimage build but to create the deb files altogether.

File: .travis.yml [hunk 1]
```diff
@@ -20,10 +20,7 @@ script:
   - ./build-scripts/build-appimage.sh
 
 after_success:
-  - find appdir -executable -type f -exec ldd {} \; | grep " => /usr" | cut -d " " -f 2-3 | sort | uniq
```
Comment (azubieta) [inline .travis.yml]: Don't drop the appimage build.

File: build-scripts/build-appimage.sh [hunk 2]
```diff
@@ -3,17 +3,19 @@
 build_scripts_dir=`dirname $0`
 source_dir=`dirname $build_scripts_dir`
 
-make DESTDIR="appdir" -j$(nproc) install
```
Comment (azubieta) [inline build-scripts/build-appimage.sh]: Don't drop the appimage build.

File: build-scripts/build-appimage.sh [hunk 3]
```diff
@@ -3,17 +3,19 @@
...
+Package: appimage-desktop-integration
+Architecture: amd64
+Maintainer: Luis Lavaire
+Depends: 
+Priority: optional
+Version: 0.1
+Description: AppImage integration with the desktop.
+" > appimage-desktop-integration/DEBIAN/control
 
-wget -c -nv "https://github.com/probonopd/linuxdeployqt/releases/download/continuous/linuxdeployqt-continuous-x86_64.AppImage" -O linuxdeployqt-continuous-x86_64.AppImage
```
Comment (azubieta) [inline build-scripts/build-appimage.sh]: Don't drop the appimage build.

File: build-scripts/build-appimage.sh [hunk 4]
```diff
@@ -3,17 +3,19 @@
...
-chmod a+x linuxdeployqt-continuous-x86_64.AppImage
+chown -R root:root appimage-desktop-integration/
 
-
-unset QTDIR; unset QT_PLUGIN_PATH ; unset LD_LIBRARY_PATH
```
Comment (azubieta) [inline build-scripts/build-appimage.sh]: Don't drop the appimage build.

File: build-scripts/build-appimage.sh [hunk 5]
```diff
@@ -3,17 +3,19 @@
...
-find appdir/
+mkdir -p appimage-desktop-integration/DEBIAN
+make DESTDIR="appimage-desktop-integration/" -j $(nproc) install
 
-rsvg-convert -f png -h 256 -w 256 ${source_dir}/res/images/appimage.svg > appdir/appimage.png
+echo "
+Package: appimage-desktop-integration
+Architecture: amd64
+Maintainer: Luis Lavaire
+Depends:
```
Comment (azubieta) [inline build-scripts/build-appimage.sh]: Missing dependencies: g++, cmake, qt

File: .travis.yml [hunk 6]
```diff
@@ -20,10 +20,7 @@ script:
...
-  - curl --upload-file Appimage_First_Run_Utility*.AppImage* https://transfer.sh/appimage-first-run-git.$(git rev-parse --short HEAD)-x86_64.AppImage
-  - wget -c https://github.com/probonopd/uploadtool/raw/master/upload.sh
-  - bash upload.sh Appimage_First_Run_Utility*.AppImage*
+  - curl --upload-file appimage-first-run.deb
```
Comment (azubieta) [inline .travis.yml]: I think that would be Ok uploading the results to transfer.sh until the new aptly repo is set

File: build-scripts/build-appimage.sh (hunk 5 above)
Comment (luis-lavaire) [reply build-scripts/build-appimage.sh]: Except for Qt5, aren't the others just dependencies during build time? We don't need them during runtime.

Comment (azubieta) [reply build-scripts/build-appimage.sh]: You are right only qt is required at runtime

File: build-scripts/build-appimage.sh (hunk 2 above)
Comment (azubieta) [reply build-scripts/build-appimage.sh]: Also, please don't build a deb file in the build-appimage script.

File: .travis.yml [hunk 7]
```diff
@@ -16,14 +16,16 @@ install:
...
+  - make -j $(nproc)
+  - sh ./build-scripts/build-appimage.sh
+  - sh ./build-scripts/build-deb.sh
 
 after_success:
   - find appdir -executable -type f -exec ldd {} \; | grep " => /usr" | cut -d " " -f 2-3 | sort | uniq
   - curl --upload-file Appimage_First_Run_Utility*.AppImage* https://transfer.sh/appimage-first-run-git.$(git rev-parse --short HEAD)-x86_64.AppImage
   - wget -c https://github.com/probonopd/uploadtool/raw/master/upload.sh
   - bash upload.sh Appimage_First_Run_Utility*.AppImage*
+  - curl --upload-file appimage-desktop-integration.deb https://transfer.sh
```
Comment (TheAssassin) [inline .travis.yml]: Does this line work? I think it needs to be `curl ... https://transfer.sh/appimage-desktop-integration.deb`.

Comment (luis-lavaire) [reply .travis.yml]: It works that way too. `transfer.sh` assigns a proper URL automatically in that way.

**Comment by azubieta** [comment]
Also, the binary package should not include all the package dependencies. This also applies for libappimage. Only the app binary and the .desktop file should be included.

**Comment by azubieta** [comment]
Implemented using cpack.

**Comment by probonopd** [comment]
Last time I tried I was unable to install due to an unmet dependency on `libappimage`.

**Comment by TheAssassin** [comment]
@probonopd the packaging is tested on Nitrux, where a libappimage package is available in the repositories.

**Comment by azubieta** [comment]
Yep, that dependency should be built at AppimageKit so this is an issue for
you :)

**Comment by TheAssassin** [comment]
@azubieta the packages are built but not published.

**Comment by probonopd** [comment]
Uh oh. I prefer to get an AppImage, exactly because I don't want to mess around with debs from various places ;-)

**Comment by azubieta** [comment]
i'm updating my repository for you to get the proper libappimage.deb

**Pull request closed without merging**